core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -149,7 +149,12 @@ private[spark] object Utils extends Logging {
 
   /** Deserialize an object using Java serialization and the given ClassLoader */
   def deserialize[T](bytes: Array[Byte], loader: ClassLoader): T = {
-    val bis = new ByteArrayInputStream(bytes)
+    deserialize(ByteBuffer.wrap(bytes), loader)
+  }
+
+  /** Deserialize an object using Java serialization and the given ClassLoader */
+  def deserialize[T](bytes: ByteBuffer, loader: ClassLoader): T = {
+    val bis = new ByteBufferInputStream(bytes)
```

Comment:
for other reviewers: I was a little worried that maybe this was (ever-so-slightly) changing a really important code path, but looks like this method actually is barely used, so not a big deal. (`SparkContext.objectFile` and `ReceivedBlockTracker.cleanupBatches`)